src/main/java/com/commercetools/sync/commons/BaseSync.java

```diff
@@ -34,32 +42,38 @@ protected BaseSync(@Nonnull final U statistics, @Nonnull final V syncOptions) {
      * this list with it's corresponding old resource in a given CTP project, and in turn it either issues update
      * actions on the existing resource if it exists or create it if it doesn't.
      *
-     * <p>The time before and after the actual sync process starts is recorded in the {@link BaseSyncStatistics}
-     * container so that the total processing time is computed in the statistics.
-     *
      * @param resourceDrafts the list of new resources as drafts.
-     * @return an instance of {@link CompletionStage&lt;U&gt;} which contains as a result an instance of {@link U} which
-     *      is a subclass of {@link BaseSyncStatistics} representing the {@code statistics} instance attribute of
-     *      {@link this} {@link BaseSync}.
+     * @return an instance of {@link CompletionStage&lt;S&gt;} which contains as a result an instance of {@link S} which
+     *      is a subclass of {@link BaseSyncStatistics} representing the {@code statistics} of this sync process.
      */
-    public CompletionStage<U> sync(@Nonnull final List<T> resourceDrafts) {
-        statistics.startTimer();
-        return process(resourceDrafts).thenApply(resultingStatistics -> {
-            resultingStatistics.calculateProcessingTime();
+    public CompletionStage<S> sync(@Nonnull final List<T> resourceDrafts) {
+        final long startTime = System.currentTimeMillis();
+        return process(resourceDrafts).thenApply(resultingStatisticsBuilder -> {
+            resultingStatisticsBuilder.setProcessingTimeInMillis(System.currentTimeMillis() - startTime);
+            final S resultingStatistics = resultingStatisticsBuilder.build();
+            updateTotalStatistics(resultingStatistics);
             return resultingStatistics;
         });
     }
 
 
     /**
-     * Returns an instance of type U which is a subclass of {@link BaseSyncStatistics} containing all the stats of the
-     * sync process; which includes a report message, the total number of update, created, failed, processed resources
-     * and the processing time of the sync in different time units and in a human readable format.
+     * Returns an instance of type S which is a subclass of {@link BaseSyncStatistics} containing the total stats of
+     * all sync processes performed by {@code this} sync instance.
      *
-     * @return a statistics object for the sync process.
+     * @return a statistics object for all sync processes performed by {@code this} sync instance.
      */
     @Nonnull
-    public U getStatistics() {
-        return statistics;
+    public synchronized S getStatistics() {
```

Comment:
since `totalStatisticsBuilder` is final  i'd rather lock on this instance, instead of whole `BaseSync` instance:
```java
public S getStatistics() {
    synchronized (totalStatisticsBuilder) {
        return totalStatisticsBuilder.build();
    }
}
```

Same for `BaseSync#updateTotalStatistics` below.

But, honestly, i don't like the whole concept of the statistic update/get:
1) do we really expect that statistic could be fetched before sync completed? If no, then we need some flag like _completed_ and return nothing (null, empty object) before completion.

2) do we expect concurrent `updateTotalStatistics()` access? As i see from `BaseSync#sync` - no. Then synchronization there is redundant (just set _completed_ flag only when the statistic is completely updated.

3) If we really should stick with concurrency here - pls provide a respective comments to the methods.